ush/generate_FV3LAM_wflow.py

```diff
@@ -744,21 +731,36 @@ def generate_FV3LAM_wflow(ushdir, logfile: str = "log.generate_FV3LAM_wflow") ->
     # If we got to this point everything was successful: move the log file to the experiment directory.
     mv_vrfy(logfile, EXPTDIR)
 
+    return EXPTDIR
 
-def setup_logging(logfile: str = "log.generate_FV3LAM_wflow") -> None:
+
+def setup_logging(logfile: str = "log.generate_FV3LAM_wflow", debug: bool = False) -> None:
     """
     Sets up logging, printing high-priority (INFO and higher) messages to screen, and printing all
     messages with detailed timing and routine info in the specified text file.
+    
+    If debug = True, print all messages to both screen and log file.
     """
-    logging.basicConfig(
-        level=logging.DEBUG,
-        format="%(name)-22s %(levelname)-8s %(message)s",
-        filename=logfile,
-        filemode="w",
-    )
+    logging.getLogger().setLevel(logging.DEBUG)
+
+    formatter = logging.Formatter("%(name)-22s %(levelname)-8s %(message)s")
+
+    fh = logging.FileHandler(logfile, mode='w')
+    fh.setLevel(logging.DEBUG)
+    fh.setFormatter(formatter)
+    logging.getLogger().addHandler(fh)
     logging.debug(f"Finished setting up debug file logging in {logfile}")
+
+    # If there are already multiple handlers, that means generate_FV3LAM_workflow was called from another function.
+    # In that case, do not change the console (print-to-screen) logging.
+    if len(logging.getLogger().handlers) > 1:
+        return
```

Comment:
I am curious, was this what solved the unittest failures?

Reply:
These changes in were needed to get log files (and messages) to behave correctly when called from another function with logging, so if that's how the unit tests are being called then I bet these changes as a whole are what fixed them. Specifically, the use of `logging.basicConfig` was the issue, as it has no effect when there already exists a logging handler. This could be solved with some settings that are only available in Python 3.8+, but this new implementation is cleaner anyway IMO.